### PR TITLE
Create new node to handle invocation in collect-clause

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
$subject
Related to #36810

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java [hunk 1]
```diff
@@ -1697,24 +1695,13 @@ public void visit(BLangExpressionStmt exprStmtNode) {
...
-        visitRestArgs(invocationExpr);
-        this.acceptNode(invocationExpr.expr);
-
-        Location pos = invocationExpr.pos;
-        if (isNilReturnInvocationInCollectClause(invocationExpr)) {
-            BLangSimpleVarRef restArg = (BLangSimpleVarRef) invocationExpr.argExprs.get(0);
-            BType invocationType = BUnionType.create(null, invocationExpr.getBType(), symTable.nilType);
+    public void visit(BLangCollectContextInvocation collectContextInvocation) {
+        BLangInvocation invocation = collectContextInvocation.invocation;
+        result = invocation = rewrite(invocation);
```
[the remark this answers is not in the document]
Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java:1700]: This affects readability and add more lines right?

[the remark this answers is not in the document]
Comment (ushirask) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java:1700]: there will be an extra line with `return` however I think the flow of the process will be more clear

[the remark this answers is not in the document]
Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java:1700]: I think the existing way is better.

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/model/TreeBuilder.java [hunk 2]
```diff
@@ -594,6 +596,9 @@ public static RecordLiteralNode.RecordVarNameFieldNode createRecordVarRefNameFie
     public static InvocationNode createInvocationNode() {
         return new BLangInvocation();
     }
+    public static CollectContextInvocationNode createCollectContextInvocationNode() {
```
Comment (rdulmina) [inline compiler/ballerina-lang/src/main/java/org/ballerinalang/model/TreeBuilder.java:599]:
```suggestion

    public static CollectContextInvocationNode createCollectContextInvocationNode() {
```

**Review by gimantha — APPROVED** [review]

**Pull request merged**